src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,188 @@
+            century: person.born ? getCentury(person.born) : undefined,
+          })),
+        );
+      })
+      .catch(() => {
+        setError('Something wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    const centuriesAsNumbers = centuries.map(Number);
+
+    // below const makes so that u chose all the existing sexes if u filter by sex or whatever
+    const filters: { condition: boolean; callback: (p: Person) => boolean }[] =
+      [
+        {
+          condition: !!sex,
+          callback: (person: Person) => person.sex === sex,
+        },
+        {
+          condition: centuries.length > 0,
+          callback: (person: Person) =>
+            person.century !== undefined &&
+            centuriesAsNumbers.includes(person.century),
+        },
+        {
+          condition: !!query,
+          callback: (person: Person) =>
+            query
+              ? person.name.toLowerCase().includes(query.toLowerCase())
+              : true,
+        },
+      ];
+
+    /*
+It starts with everyone in the list. For each filter rule, 
+if that rule is "turned on," it removes anyone who doesn't match. By the end, 
+you’re left with only the people who pass all the chosen filters.
+*/
+
+    return filters.reduce((acc, { condition, callback }) => {
+      if (condition) {
+        return acc.filter(callback);
+      }
+
+      return acc;
+    }, people);
+  }, [people, centuries, query, sex]);
+
+  const sortedFilteredPeople = useMemo(() => {
+    if (!sortColumn) {
+      return filteredPeople;
+    }
+    //
+
+    //here it does the filtering
+    const comparePeople = (a: Person, b: Person) => {
+      const valueA = a[sortColumn];
+      const valueB = b[sortColumn];
+
+      if (valueA === undefined && valueB === undefined) {
+        return 0;
+      }
+
+      if (valueA === undefined) {
+        return 1;
+      }
+
+      if (valueB === undefined) {
+        return -1;
+      }
+
+      if (sortDirection) {
+        return valueA > valueB ? -1 : valueA < valueB ? 1 : 0;
+      } else {
+        return valueA < valueB ? -1 : valueA < valueB ? 1 : 0;
+      }
+    };
+
+    return [...filteredPeople].sort(comparePeople);
+  }, [filteredPeople, sortColumn, sortDirection]);
+
+  //////////////////////
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchParams(newParams => {
+      newParams.set('query', event.target.value);
+
+      return newParams;
+    });
+  };
+
+  const peopleMap: PeopleMap = useMemo(
+    () =>
+      people.reduce(
+        (map, person) => ({
+          ...map,
+          [person.name]: person,
+        }),
+        {},
+      ),
+    [people],
+  );
+
+  const toggleArrow = (column: string) => {
+    const [order, sort] =
+      column === sortColumn
+        ? sortDirection
+          ? [null, null]
+          : ['desc', column]
+        : [null, column];
```

Comment:
The `toggleArrow` function logic might not be correctly setting the `order` and `sort` parameters. Ensure that the logic correctly toggles between ascending, descending, and no sort states.